packages/kolme-store/src/error.rs

```diff
@@ -20,14 +41,53 @@ pub enum KolmeStoreError {
     // kolme#144 - Reports a double insert (Block already exists with same hash and insert)
     #[error("Block already in database: {height}")]
     MatchingBlockAlreadyInserted { height: u64 },
+
     #[error("Transaction is already present in database: {txhash}")]
     TxAlreadyInDb { txhash: Sha256Hash },
+
+    #[error("get_height_for_tx: invalid height bytes in Fjall store for tx {txhash:?}, bytes: {bytes:?}, reason: {reason}")]
+    InvalidHeightInFjall {
+        txhash: Sha256Hash,
+        bytes: Vec<u8>,
+        reason: String,
+    },
+
+    #[error("Merkle validation error: child hash {child} not found")]
+    MissingMerkleChild { child: Sha256Hash },
+
+    #[error("Invalid genesis message count in first block")]
+    InvalidGenesisMessageCount,
+
+    #[error("Invalid message type in first block: expected Genesis")]
+    InvalidFirstBlockMessageType,
+
+    #[error("Fjall error: {0}")]
+    Fjall(String),
+
     #[error("{0}")]
     Other(String),
```

Comment:
Is it possible to remove this variant?  The two places I see it used are the `From<anyhow::Error>` impl below (see comment there), a couple of places where I think `Custom` might work.